From 04b412e10d7176764490121b621d0d3be347b31a Mon Sep 17 00:00:00 2001
From: Issam Maghni <concatime@users.noreply.github.com>
Date: Thu, 6 May 2021 19:26:42 +0000
Subject: [PATCH] Add missing second equal symbol for comparaison

Due to this bug, the `if` statement is always entered.
---
 pam_rundir.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Patch-Source: https://github.com/jjk-jacky/pam_rundir/pull/5

diff --git a/pam_rundir.c b/pam_rundir.c
index 2d2c201..67c7ca6 100644
--- a/pam_rundir.c
+++ b/pam_rundir.c
@@ -197,7 +197,7 @@ write_counter (int fd, int count)
             r = write (fd, buf + w, l - w);
             if (r < 0)
             {
-                if (errno = EINTR)
+                if (errno == EINTR)
                     continue;
                 if (w > 0)
                     emergency_invalidate_counter (fd);
